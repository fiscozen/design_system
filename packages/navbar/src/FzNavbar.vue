<script lang="ts" setup>
import { ref, onBeforeMount, onMounted, onUnmounted, computed } from 'vue'
import { FzIconButton } from '@fiscozen/button'
import { FzNavbarEmits, FzNavbarProps } from './types'

const props = withDefaults(defineProps<FzNavbarProps>(), {
  variant: 'horizontal'
})

const emit = defineEmits<FzNavbarEmits>()

const width = ref(0)

const onResize = () => {
  width.value = window.innerWidth
}
onBeforeMount(() => {
  onResize()
})

onMounted(() => {
  window.addEventListener('resize', onResize)
})
onUnmounted(() => {
  window.removeEventListener('resize', onResize)
})

const isMobile = computed(() => Boolean(width.value <= 1024))
const isVertical = computed(() => Boolean(props.variant === 'vertical'))
const isHorizontal = computed(() => Boolean(props.variant === 'horizontal'))
</script>

<template>
  <header
    class="flex p-12 shadow"
    :class="{
      'justify-between': isMobile,
      'h-full w-56 flex-col': isVertical && !isMobile,
      'h-56 w-full': isHorizontal || isMobile
    }"
  >
    <template v-if="!isMobile">
      <div :class="{ 'mr-32': isHorizontal, 'mb-32': isVertical }">
        <slot name="brand-logo" :isMobile :isHorizontal :isVertical></slot>
      </div>
      <div class="flex gap-4" :class="{ 'flex-row': isHorizontal, 'flex-col': isVertical }">
        <slot name="navigation" :isVertical :isHorizontal :isMobile></slot>
      </div>
      <div
        class="flex gap-16"
        :class="{ 'ml-auto flex-row': isHorizontal, 'mt-auto flex-col': isVertical }"
      >
        <slot name="notifications" :isHorizontal :isVertical :isMobile></slot>
        <slot name="user-menu" :isHorizontal :isMobile :isVertical></slot>
      </div>
    </template>
    <template v-else>
      <FzIconButton
        iconName="bars"
        variant="secondary"
        tooltip="menu"
        @click="emit('fznavbar:menuButtonClick')"
      />
      <div>
        <slot name="brand-logo" :isMobile :isHorizontal :isVertical></slot>
      </div>
      <div>
        <slot
          :name="isHorizontal ? 'notifications' : 'user-menu'"
          :isHorizontal
          :isVertical
          :isMobile
        ></slot>
      </div>
    </template>
  </header>
</template>

<style></style>
