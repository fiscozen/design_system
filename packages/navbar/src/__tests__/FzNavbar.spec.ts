import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import FzNavbar from '../FzNavbar.vue'

describe('FzNavbar', () => {
  it('should match snapshot', () => {
    const wrapper = mount(FzNavbar, {})

    expect(wrapper.html()).toMatchSnapshot()
  })
})
