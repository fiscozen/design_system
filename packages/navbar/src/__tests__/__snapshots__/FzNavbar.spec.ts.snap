// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzNavbar > should match snapshot in horizontal layout large screen 1`] = `
"<header class="flex px-12 py-12 shadow h-56 w-full">
  <div class="mr-32">
    <div id="brandlogo" class="desktop"></div>
  </div>
  <div class="flex gap-4 flex-row">
    <div class="link">one</div>
    <div class="link">two</div>
    <div class="link">three</div>
  </div>
  <div class="flex gap-16 ml-auto flex-row">
    <div id="notification"></div>
    <div id="avatar"></div>
  </div>
</header>"
`;

exports[`FzNavbar > should match snapshot in horizontal layout small screen 1`] = `
"<header class="flex px-12 py-12 shadow justify-between h-56 w-full"><button type="button" class="relative relative rounded flex flex-shrink items-center justify-center w-32 h-32 focus:border-blue-600 focus:border-solid focus:border-1 text-grey-500 bg-core-white border-1 border-grey-200 hover:bg-grey-100 disabled:text-grey-100"><svg class="svg-inline--fa fa-bars fa-lg" aria-hidden="true" focusable="false" data-prefix="fasl" data-icon="bars" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
      <path class="" fill="currentColor" d="M0 64H448V96H0V64zM0 224H448v32H0V224zM448 384v32H0V384H448z"></path>
    </svg>
    <!--v-if--><span class="hidden h-0 w-0">menu</span>
  </button>
  <div>
    <div id="brandlogo" class="mobile"></div>
  </div>
  <div>
    <div id="notification"></div>
  </div>
</header>"
`;

exports[`FzNavbar > should match snapshot in vertical layout large screen 1`] = `
"<header class="flex px-12 py-12 shadow h-full w-56 flex-col">
  <div class="mb-32">
    <div id="brandlogo" class="desktop"></div>
  </div>
  <div class="flex gap-4 flex-col">
    <div class="link">one</div>
    <div class="link">two</div>
    <div class="link">three</div>
  </div>
  <div class="flex gap-16 mt-auto flex-col">
    <div id="notification"></div>
    <div id="avatar"></div>
  </div>
</header>"
`;

exports[`FzNavbar > should match snapshot in vertical layout small screen 1`] = `
"<header class="flex px-12 py-12 shadow justify-between h-56 w-full"><button type="button" class="relative relative rounded flex flex-shrink items-center justify-center w-32 h-32 focus:border-blue-600 focus:border-solid focus:border-1 text-grey-500 bg-core-white border-1 border-grey-200 hover:bg-grey-100 disabled:text-grey-100"><svg class="svg-inline--fa fa-bars fa-lg" aria-hidden="true" focusable="false" data-prefix="fasl" data-icon="bars" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
      <path class="" fill="currentColor" d="M0 64H448V96H0V64zM0 224H448v32H0V224zM448 384v32H0V384H448z"></path>
    </svg>
    <!--v-if--><span class="hidden h-0 w-0">menu</span>
  </button>
  <div>
    <div id="brandlogo" class="mobile"></div>
  </div>
  <div>
    <div id="avatar"></div>
  </div>
</header>"
`;
